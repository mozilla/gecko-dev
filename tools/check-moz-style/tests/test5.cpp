// License bogus

// runtime/virtual
class ShouldHaveVirtualDes {
  virtual foo();
};

int main() {
  // runtime/memset
  memset(blah, sizeof(blah), 0);

  // runtime/rtti
  dynamic_cast<Derived*>(obj);

  // runtime/sizeof
  int varname = 0;
  int mySize = sizeof(int);

  // runtime/threadsafe_fn
  getpwuid();
  strtok();

  return 0;
}
