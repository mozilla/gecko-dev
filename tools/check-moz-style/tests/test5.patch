# Test
Bug 12 - patch summary with no bug number

Some bogus patch description
diff --git a/tests/test5.cpp b/tests/test5.cpp
new file mode 100644
--- /dev/null
+++ b/tests/test5.cpp
@@ -0,0 +1,24 @@
+// License bogus
+
+// runtime/virtual
+class ShouldHaveVirtualDes {
+  virtual foo();
+};
+
+int main() {
+  // runtime/memset
+  memset(blah, sizeof(blah), 0);
+
+  // runtime/rtti
+  dynamic_cast<Derived*>(obj);
+
+  // runtime/sizeof
+  int varname = 0;
+  int mySize = sizeof(int);
+
+  // runtime/threadsafe_fn
+  getpwuid();
+  strtok();
+
+  return 0;
+}
