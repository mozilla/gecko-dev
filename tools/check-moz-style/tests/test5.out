tests/test5.cpp:4:  The class ShouldHaveVirtualDes probably needs a virtual destructor due to having virtual method(s), one declared at line 5.  [runtime/virtual] [4]
tests/test5.cpp:10:  Did you mean "memset(blah, 0, sizeof(blah))"?  [runtime/memset] [4]
tests/test5.cpp:13:  Do not use dynamic_cast<>.  If you need to cast within a class hierarchy, use static_cast<> to upcast.  Mozilla doesn't support RTTI.  [runtime/rtti] [5]
tests/test5.cpp:17:  Using sizeof(type).  Use sizeof(varname) instead if possible  [runtime/sizeof] [1]
tests/test5.cpp:20:  Consider using getpwuid_r(...) instead of getpwuid(...) for improved thread safety.  [runtime/threadsafe_fn] [2]
tests/test5.cpp:21:  Consider using strtok_r(...) instead of strtok(...) for improved thread safety.  [runtime/threadsafe_fn] [2]
Done processing tests/test5.cpp
